src/promptflow/promptflow/_sdk/_constants.py

```diff
@@ -2,11 +2,60 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
+import os
 from enum import Enum
 from pathlib import Path
 
 LOGGER_NAME = "promptflow"
 
+PROMPT_FLOW_HOME_DIR_ENV_VAR = "PF_HOME_DIRECTORY"
+PROMPT_FLOW_DIR_NAME = ".promptflow"
+
+
+def _prepare_home_dir() -> Path:
+    """Prepare prompt flow home directory.
+
+    User can configure it by setting environment variable: `PF_HOME_DIRECTORY`;
+    if not configured, or configured value is not valid, use default value: "~/.promptflow/".
+    """
+    from promptflow._utils.logger_utils import LoggerFactory
+
+    logger = LoggerFactory.get_logger(LOGGER_NAME)
+
+    if PROMPT_FLOW_HOME_DIR_ENV_VAR in os.environ:
+        logger.debug(
+            f"environment variable {PROMPT_FLOW_HOME_DIR_ENV_VAR!r} is set, honor it preparing home directory."
+        )
+        try:
+            pf_home_dir = Path(os.getenv(PROMPT_FLOW_HOME_DIR_ENV_VAR)).resolve()
+            pf_home_dir.mkdir(parents=True, exist_ok=True)
+            return pf_home_dir
+        except Exception as e:  # pylint: disable=broad-except
+            _warning_message = (
+                "Invalid configuration for prompt flow home directory: "
+                f"{os.getenv(PROMPT_FLOW_HOME_DIR_ENV_VAR)!r}: {str(e)!r}.\n"
+                'Fall back to use default value: "~/.promptflow/".'
+            )
+            logger.warning(_warning_message)
+
+    try:
+        logger.debug("preparing home directory with default value.")
+        pf_home_dir = (Path.home() / PROMPT_FLOW_DIR_NAME).resolve()
```

Comment:
why we need to have two  try catch?

Reply:
the second try for default value is for better error message for edge cases, e.g. no write permissino to operate `~/.promptflow/`, otherwise user may see a traceback and less user friendly

Reply:
I see, you want to do a fall back logic. Looks good to me.